src/cursor/cur_join.c

```diff
@@ -482,8 +483,10 @@ __curjoin_entry_in_range(
         }
 
         if (!passed) {
-            if (iter != NULL && (iter->is_equal || F_ISSET(end, WT_CURJOIN_END_LT)))
+            if (iter != NULL && (iter->is_equal || F_ISSET(end, WT_CURJOIN_END_LT))) {
+                WT_RET(__curjoin_iter_bump(iter));
```

Comment:
Minor suggestion:  Maybe add a comment here similar to the one at the start of `curjoin_iter_bump` so somebody reading this code knows whey we're making the call.

Reply:
Thanks, done.